lib/spack/spack/schema/env.py

```diff
@@ -53,6 +53,7 @@
                         },
                     },
                     "specs": spack.schema.spec_list_schema,
+                    "detect-changes-with-git": {"type": "boolean", "default": False},
```

Comment:
1. Why `False` by default, or why even have an option here?  I think it would be better if we just default to git if a `develop` directory is cloned with `git`.
2. If there is a config option for this we shouldn't put it in the env schema -- let's put it in `config.yaml` (so that it is in an actual config section)... but I'm not sure why it can't just be automatic.

Reply:
This only detects changes for tracked files: if the user adds a new file without calling `git add`, then this feature would fail to detect a change. IMO that meant this should be off by default (since the user has to be aware of a caveat to know when this will work).

Reply:
(note, this conversation was accidentally continued at https://github.com/spack/spack/pull/41013#discussion_r1431081172)